.github/workflows/Docker-image-to-package.yml

```diff
@@ -0,0 +1,43 @@
+name: Create and publish a Docker image to ghcr.io
+
+on:
+  push:
+     branches: [ "main" ]
+  pull_request:
+     branches: [ "main" ]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
```

Comment:
Is that hash value needed ?

Reply:
The _docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9_ can be replaced with _login-action@v2_